File: Ent_UsAdmin/grupos.vb
Imports System.Data.OleDb

Public Class grupos
    Dim strgrupos As String
    Private Sub Panel1_MouseDown(sender As Object, e As MouseEventArgs) Handles Panel1.MouseDown
        If e.Button = Windows.Forms.MouseButtons.Left And Me.WindowState <> FormWindowState.Maximized Then
            MoveForm(Me)
        End If
    End Sub






    Private Sub grupos_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        iniciarconexionR()
        Combofiltro.Items.Add("Todos")
        Combofiltro.Items.Add("Con Nombre")
        Combofiltro.Items.Add("Con Código")
        Combofiltro.SelectedIndex = 0
    End Sub





    Private Sub Combofiltro_SelectedIndexChanged(sender As Object, e As EventArgs) Handles Combofiltro.SelectedIndexChanged
        If Combofiltro.Text = "Todos" Then
            strgrupos = "select id,nm,cdg from grp"
            cargargrupos()

        End If
    End Sub

    Public Sub cargargrupos()
        dtgrupos.Rows.Clear()

        Dim ejec = New OleDbCommand(strgrupos)
        ejec.Connection = conexionempresaR
        Dim nombre, codigo, id As String

        Dim myreadergrupos As OleDbDataReader = ejec.ExecuteReader()
        While myreadergrupos.Read
            id = myreadergrupos("id")
            nombre = myreadergrupos("nm")
            codigo = myreadergrupos("cdg")
            dtgrupos.Rows.Add(id, codigo, nombre)
        End While
        myreadergrupos.Close()

    End Sub

    
   
    Private Sub btn_agregar_Click(sender As Object, e As EventArgs) Handles btn_agregar.Click
        agregargrupos.Show()

    End Sub

    Private Sub dtgrupos_CellContentClick(sender As Object, e As DataGridViewCellEventArgs) Handles dtgrupos.CellContentClick

    End Sub

    Private Sub dtgrupos_CellContentDoubleClick(sender As Object, e As DataGridViewCellEventArgs) Handles dtgrupos.CellContentDoubleClick
        PermisosGrupo.idgrupousuarios = dtgrupos.Rows(dtgrupos.CurrentRow.Index).Cells(0).Value
        PermisosGrupo.Show()
    End Sub
End Class

File: Ent_UsAdmin/Catalogos/grupos.cs
using System;
using System.Data.OleDb;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

namespace ConfiaAdmin
{

    public partial class grupos
    {
        private string strgrupos;

        public grupos()
        {
            InitializeComponent();
        }
        private void Panel1_MouseDown(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Left & WindowState != FormWindowState.Maximized)
            {
                Module_resize.MoveForm(this);
            }
        }






        private void grupos_Load(object sender, EventArgs e)
        {
            Module1.iniciarconexionR();
            Combofiltro.Items.Add("Todos");
            Combofiltro.Items.Add("Con Nombre");
            Combofiltro.Items.Add("Con Código");
            Combofiltro.SelectedIndex = 0;
        }





        private void Combofiltro_SelectedIndexChanged(object sender, EventArgs e)
        {
            if (Combofiltro.Text == "Todos")
            {
                strgrupos = "select id,nm,cdg from grp";
                cargargrupos();

            }
        }

        public void cargargrupos()
        {
            dtgrupos.Rows.Clear();

            var ejec = new OleDbCommand(strgrupos);
            ejec.Connection = Module1.conexionempresaR;
            string nombre, codigo, id;

            var myreadergrupos = ejec.ExecuteReader();
            while (myreadergrupos.Read())
            {
                id = Conversions.ToString(myreadergrupos["id"]);
                nombre = Conversions.ToString(myreadergrupos["nm"]);
                codigo = Conversions.ToString(myreadergrupos["cdg"]);
                dtgrupos.Rows.Add(id, codigo, nombre);
            }
            myreadergrupos.Close();

        }



        private void btn_agregar_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.agregargrupos.Show();

        }

        private void dtgrupos_CellContentClick(object sender, DataGridViewCellEventArgs e)
        {

        }

        private void dtgrupos_CellContentDoubleClick(object sender, DataGridViewCellEventArgs e)
        {
            My.MyProject.Forms.PermisosGrupo.idgrupousuarios = Conversions.ToInteger(dtgrupos.Rows[dtgrupos.CurrentRow.Index].Cells[0].Value);
            My.MyProject.Forms.PermisosGrupo.Show();
        }
    }
}